[LFv1] List invited users on user management page

Nateowami opened the pull request:
These are fairly straightforward changes. On the back end it adds an invitees list to the user management DTO. On the front end it lists them after active users on the user management page.

This UI has the disadvantage of listing email under the username header when no username is available.

![Screenshot from 2019-04-03 21-31-05](https://user-images.githubusercontent.com/6140710/55523630-07d4ad00-5658-11e9-8069-84febedafc21.png)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/web-languageforge/645)
<!-- Reviewable:end -->

megahirt commented:
@Nateowami Can you rebase this branch off the latest master?  Ira and I think that unit tests are failing because this branch was resurrected from a while ago.  Also see Ira's comment in the trello card about using the isInvited flag on the UserModel.  That is a recent addition since end of January.  Sorry I didn't mention that before - forgot about that piece.

megahirt commented:
The UI looks good, Nathaniel.  Showing the email address under email
address is not bad - in fact it should be the only information we have
since invited users haven't picked a username yet.

On Thu, Apr 4, 2019 at 8:34 AM Nathaniel Paulus <notifications@github.com>
wrote:

> @Nateowami <https://github.com/Nateowami> requested your review on: #645
> <https://github.com/sillsdev/web-languageforge/pull/645> List invited
> users on user management page.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/sillsdev/web-languageforge/pull/645#event-2251532225>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADSPKW140ZQd4r6Uws7-qOWCgHgouCWBks5vdVZfgaJpZM4cbv37>
> .
>

Nateowami commented:
@megahirt Will do. Sorry I forgot to run the unit tests.

Nateowami commented:
@megahirt I've rebased on master, changed the logic to use the `isInvited` flag, and gotten the tests passing.

As a consequence of using the `isInvited` flag to distinguish between invited and regular users, it no longer makes sense to use the presence of a username as a qualification for being a "regular" user. My thinking is that the categories of "invited" and "regular" users ought to be mutually exclusive and collectively exhaustive (every user falls into exactly one of the two categories). This necessitated updating a test that assumed otherwise.

It's possible old invited users will not show up in the list of invited users, if they were created before the change to having the `isInvited` flag. I think I had some users disappear from the list when I changed the logic, making me think there isn't a database migration.

irahopkinson commented:
We don’t have existing code that migrates the database on the fly. We have
one-off code that migrates the database at one time. The live and QA
database was migrated when the `isInvited` flag was introduced. If you
haven’t run that migration on you local database I would expect you would
see the problem of them not being mutually exclusive in your data.

I would recommend you run the migration locally before merging to satisfy
yourself it is working. See this card for migration
https://trello.com/c/C8pyYfU4 it’s just a mongoDB query.

Blessings, Ira

On Sat, 13 Apr 2019 at 6:45 AM, Nathaniel Paulus <notifications@github.com>
wrote:

> @megahirt <https://github.com/megahirt> I've rebased on master, changed
> the logic to use the isInvited flag, and gotten the tests passing.
>
> As a consequence of using the isInvited flag to distinguish between
> invited and regular users, it no longer makes sense to use the presence of
> a username as a qualification for being a "regular" user. My thinking is
> that the categories of "invited" and "regular" users ought to be mutually
> exclusive and collectively exhaustive (every user falls into exactly one of
> the two categories). This necessitated updating a test that assumed
> otherwise.
>
> It's possible old invited users will not show up in the list of invited
> users, if they were created before the change to having the isInvited
> flag. I think I had some users disappear from the list when I changed the
> logic, making me think there isn't a database migration.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sillsdev/web-languageforge/pull/645#issuecomment-482754465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFUtSRv1GGc63Hp_zIXuI3hA147XUUXIks5vgRqNgaJpZM4cbv37>
> .
>
-- 
Ira Hopkinson

SIL International Language Software Development, PaLaSo <http://palaso.org/>

megahirt commented:
I will be doing testing on this change to ensure that the isInvitee flag is being properly set and that the flag is set on all users without an username

irahopkinson commented:
@megahirt I think that will get you the wrong results. As I mentioned above see the [migration card](https://trello.com/c/C8pyYfU4) for criteria for the `isInvited` flag - basically _null_ or _false_ for the `active` flag and no `password`. The `username` may not be set for other reasons like if you signed up with a google account - its not a valid test for an invited user.

megahirt commented:
Good call, Ira.  I will make sure and test based upon the criteria in the
migration card.

On Mon, Apr 15, 2019 at 3:55 PM Ira Hopkinson <notifications@github.com>
wrote:

> @megahirt <https://github.com/megahirt> I think that will get you the
> wrong results. As I mentioned above see the migration card
> <https://trello.com/c/C8pyYfU4> for criteria for the isInvited flag -
> basically *null* or *false* for the active flag and no password. The
> username may not be set for other reasons like if you signed up with a
> google account - its not a valid test for an invited user.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sillsdev/web-languageforge/pull/645#issuecomment-483166322>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADSPKYHX38EIb0rIX20W_TdmoLwbnbpfks5vhD5PgaJpZM4cbv37>
> .
>
